recipes/openai_textcat.py

```diff
@@ -0,0 +1,287 @@
+        example["meta"]["reason"] = output["reason"]
+        example["answer"] = "accept"
+        # Filter removes any empty strings
+        example["accept"] = list(
+            filter(
+                None, [normalize_label(s.strip()) for s in output["answer"].split(",")]
+            )
+        )
+        return example
+
+
+@prodigy.recipe(
+    # fmt: off
+    "textcat.openai.correct",
+    input_path=("Path to jsonl data to annotate", "positional", None, Path),
+    prompt_path=("Path to jinja2 prompt template", "positional", None, Path),
+    labels=("Labels (comma delimited)", "option", "L", lambda s: s.split(",")),
+    lang=("Language to initialize spaCy model", "option", "l", str),
+    model=("GPT-3 model to use for completion", "option", "m", str),
+    batch_size=("Batch size to send to OpenAI API", "option", "b", int),
+    segment=("Split sentences", "flag", "S", bool),
+    examples_path=("Examples file to help define the task", "option", "e", Path),
+    max_examples=("Max examples to include in prompt", "option", "n", int),
+    exclusive_classes=("Make the classification task exclusive", "flag", "E", bool),
+    verbose=("Print extra information to terminal", "flag", "v", bool),
+    # fmt: on
+)
+def textcat_openai_correct(
+    dataset: str,
+    input_path: Path,
+    prompt_path: Path,
+    labels: List[str] = None,
+    lang: str = "en",
+    model: str = "text-davinci-003",
+    batch_size: int = 10,
+    segment: bool = False,
+    examples_path: Optional[Path] = None,
+    max_examples: int = 2,
+    exclusive_classes: bool = False,
+    verbose: bool = False,
+):
+    api_key, api_org = get_api_credentials(model)
+    examples = read_prompt_examples(examples_path, example_class=TextCatPromptExample)
+    if labels is None:
+        msg.fail("textcat.teach requires at least one --label", exits=1)
+    nlp = spacy.blank(lang)
+
+    if segment:
+        nlp.add_pipe("sentencizer")
+
+    # Create partial render of the template
+    if not exclusive_classes and len(labels) == 1:
+        msg.warn(
+            "Binary classification should always be exclusive. Setting "
+            "`exclusive_classes` parameter to True"
+        )
+        exclusive_classes = True
+
+    template = Template(
+        load_template(prompt_path).render(
+            exclusive_classes=exclusive_classes, labels=labels, examples=examples
+        )
+    )
+
+    # Create OpenAISuggester with ChatGPT parameters
+    openai = TextCatOpenAISuggester(
+        prompt_template=template,
+        labels=labels,
+        max_examples=max_examples,
+        segment=segment,
+        openai_api_org=api_org,
+        openai_api_key=api_key,
+        openai_model=model,
+        openai_timeout_s=30,
+        openai_n=10,
+        verbose=verbose,
+    )
+    for eg in examples:
+        openai.add_example(eg)
+
+    # Set up the stream
+    stream = list(srsly.read_jsonl(input_path))
```

Comment:
Technically, the user might provide a CSV as well here, right?

The [`get_stream`](https://prodi.gy/docs/api-components#get_stream) function would be better here.

Reply:
Noted on this!